### PR TITLE
rocm-related: misc changes

### PR DESCRIPTION
**Pull request by Madouura**
###### Description of changes
Tracking: #197885
Supercedes: #203335

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Madouura** [comment]
Please do review, but do not merge until I figure out what's going on with ``nixpkgs-review``.
This is a new error now only showing and I suspect it might have to do with what SuperSandro2000 commented on about ``strictDeps``.

File: pkgs/development/libraries/rocwmma/default.nix [hunk 1]
```diff
@@ -6,31 +6,24 @@
...
-assert buildTests -> gtest != null;
-assert buildTests -> rocblas != null;
-assert buildDocs -> texlive != null;
-assert buildDocs -> doxygen != null;
-assert buildDocs -> sphinx != null;
-assert buildDocs -> python3Packages != null;
-
 # Building tests isn't working for now
 # undefined reference to symbol '_ZTIN7testing4TestE'
 assert buildTests == false;
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/rocwmma/default.nix:23]: I think this should be moved to meta.broken

Comment (Madouura) [reply pkgs/development/libraries/rocwmma/default.nix:23]: Because the tests don't work?
You mean something like ``broken = (!buildTests)``?

**Comment by Madouura** [comment]
Result of `nixpkgs-review pr 203412` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>25 packages built:</summary>
  <ul>
    <li>blender-hip</li>
    <li>clang-ocl</li>
    <li>composable_kernel</li>
    <li>hip</li>
    <li>hipcub</li>
    <li>hipsparse</li>
    <li>llvmPackages_rocm.clang</li>
    <li>llvmPackages_rocm.clangNoCompilerRt</li>
    <li>llvmPackages_rocm.llvm</li>
    <li>miopengemm</li>
    <li>rccl</li>
    <li>rocblas</li>
    <li>rocclr</li>
    <li>rocm-comgr</li>
    <li>rocm-device-libs</li>
    <li>rocm-opencl-icd</li>
    <li>rocm-opencl-runtime</li>
    <li>rocm-runtime</li>
    <li>rocminfo</li>
    <li>rocmlir</li>
    <li>rocprim</li>
    <li>rocrand</li>
    <li>rocsparse</li>
    <li>rocthrust</li>
    <li>rocwmma</li>
  </ul>
</details>

**Comment by Madouura** [comment]
One moment while I deprecate ``rocmVersion`` and ``repoVersion``...

**Comment by Madouura** [comment]
It'll be a bit to compile everything again, but unless there are other issues, this should be good for review and merge.

**Comment by Madouura** [comment]
Result of `nixpkgs-review pr 203412` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>26 packages built:</summary>
  <ul>
    <li>blender-hip</li>
    <li>clang-ocl</li>
    <li>composable_kernel</li>
    <li>hip</li>
    <li>hipcub</li>
    <li>hipsparse</li>
    <li>llvmPackages_rocm.clang</li>
    <li>llvmPackages_rocm.clangNoCompilerRt</li>
    <li>llvmPackages_rocm.llvm</li>
    <li>miopengemm</li>
    <li>rccl</li>
    <li>rocblas</li>
    <li>rocclr</li>
    <li>rocm-comgr</li>
    <li>rocm-device-libs</li>
    <li>rocm-opencl-icd</li>
    <li>rocm-opencl-runtime</li>
    <li>rocm-runtime</li>
    <li>rocminfo</li>
    <li>rocmlir</li>
    <li>rocprim</li>
    <li>rocrand</li>
    <li>rocsparse</li>
    <li>rocthrust</li>
    <li>rocwmma</li>
    <li>tensile</li>
  </ul>
</details>

**Review by Flakebi — COMMENTED** [review]
Looks good from my side

File: pkgs/development/libraries/rocwmma/default.nix (hunk 1 above)
Comment (SuperSandro2000) [reply pkgs/development/libraries/rocwmma/default.nix]: yes

**Pull request merged**